internal/core/src/query/SearchOnGrowing.cpp

```diff
@@ -94,15 +94,20 @@ SearchOnGrowing(const segcore::SegmentGrowingImpl& segment,
         FloatSegmentIndexSearch(
             segment, info, query_data, num_queries, bitset, search_result);
     } else {
+        std::shared_lock<std::shared_mutex> read_chunk_mutex(
+            segment.get_chunk_mutex());
+        // check SyncDataWithIndex() again, in case the vector chunks has been removed.
+        if (segment.get_indexing_record().SyncDataWithIndex(field.get_id())) {
+            FloatSegmentIndexSearch(
```

Comment:
return here? it seems that no need to run the following broute force

Reply:
updated